relayer/workers/ethrelayer/main.go

```diff
@@ -52,23 +51,18 @@ func (w *Worker) Start(ctx context.Context, eg *errgroup.Group) error {
 		return nil
 	})
 
-	// channel for messages from ethereum
-	ethMessages := make(chan []chain.Message, 1)
-	// channel for headers from ethereum (it's a blocking channel so that we
-	// can guarantee that a header is forwarded before we send dependent messages)
-	ethHeaders := make(chan chain.Header)
+	// channel for payloads from ethereum
+	payloads := make(chan ParachainPayload, 1)
```

Comment:
My feeling is that, if we make `payloads` unbuffered, we should probably not be using a channel here at all. I think we should give this option serious consideration

Reply:
agree but lets ship it and focus on other priorities, as i think this worker is fairly robust and reliable for now

Reply:
@Rizziepit perhaps, though I think unbuffered channels + goroutines are still a good for implementing a clean separation of concerns between a producer and a consumer.

Kind of like cooperative multitasking using generators (like in Python), except since Golang doesn't have those we have to use unbuffered channels + goroutines.